src/windows-emulator/minidump_loader.cpp

```diff
@@ -0,0 +1,758 @@
+        stats.handle_count = dump_file->handles().size();
+        stats.has_exception = (dump_file->get_exception_info() != nullptr);
+        stats.has_system_info = (dump_file->get_system_info() != nullptr);
+
+        for (const auto& segment : dump_file->memory_segments())
+        {
+            stats.total_memory_size += segment.size;
+        }
+
+        win_emu.log.info(
+            "Summary: %s, %zu threads, %zu modules, %zu regions, %zu segments, %zu handles, %" PRIu64 " bytes memory\n",
+            get_architecture_string(dump_file).c_str(), stats.thread_count, stats.module_count,
+            stats.memory_region_count, stats.memory_segment_count, stats.handle_count, stats.total_memory_size);
+    }
+
+    void process_streams(windows_emulator& win_emu, const minidump::minidump_file* dump_file)
+    {
+        if (!dump_file)
+        {
+            return;
+        }
+
+        // Process system info
+        const auto* sys_info = dump_file->get_system_info();
+        if (sys_info)
+        {
+            win_emu.log.info("System: OS %u.%u.%u, %u processors, type %u, platform %u\n", sys_info->major_version,
+                             sys_info->minor_version, sys_info->build_number, sys_info->number_of_processors,
+                             sys_info->product_type, sys_info->platform_id);
+        }
+
+        // Process memory info
+        const auto& memory_regions = dump_file->memory_regions();
+        uint64_t total_reserved = 0;
+        uint64_t total_committed = 0;
+        size_t guard_pages = 0;
+        for (const auto& region : memory_regions)
+        {
+            total_reserved += region.region_size;
+            if (region.state & MEM_COMMIT)
+            {
+                total_committed += region.region_size;
+            }
+            if (region.protect & PAGE_GUARD)
+            {
+                guard_pages++;
+            }
+        }
+        win_emu.log.info("Memory: %zu regions, %" PRIu64 " bytes reserved, %" PRIu64
+                         " bytes committed, %zu guard pages\n",
+                         memory_regions.size(), total_reserved, total_committed, guard_pages);
+
+        // Process memory content
+        const auto& memory_segments = dump_file->memory_segments();
+        uint64_t min_addr = UINT64_MAX;
+        uint64_t max_addr = 0;
+        for (const auto& segment : memory_segments)
+        {
+            min_addr = std::min(min_addr, segment.start_virtual_address);
+            max_addr = std::max(max_addr, segment.end_virtual_address());
+        }
+        if (!memory_segments.empty())
+        {
+            win_emu.log.info("Content: %zu segments, range 0x%" PRIx64 "-0x%" PRIx64 " (%" PRIu64 " bytes span)\n",
+                             memory_segments.size(), min_addr, max_addr, max_addr - min_addr);
+        }
+
+        // Process modules
+        const auto& modules = dump_file->modules();
+        for (const auto& module : modules)
```

Comment:
module is reserved, not gonna highlight it anymore, might make sense to just replace all module variables in this file :P